### PR TITLE
[Improve][Connector-V2] Support user-defined schema for reading text file

### LINKED ISSUE #2952: [Improve][Connector-V2] Support user-defined schema when using file connector read text files
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

In MVP version file connector only support read text files line by line, not support using user-defined schema parse data from content, this pr will implement this feature.

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by TyrantLucifer**
## Purpose of this pull request

close #2952 

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: docs/en/connector-v2/source/FtpFile.md [hunk 1]
```diff
@@ -51,6 +55,38 @@ The target ftp password is required
 
 The source file path.
 
+### delimiter [string]
+
+Field delimiter, used to tell connector how to slice and dice fields when reading text files
```
Comment (ashulin) [inline docs/en/connector-v2/source/FtpFile.md:61]: Add '\001' code description.

Comment (TyrantLucifer) [reply docs/en/connector-v2/source/FtpFile.md:61]: Done, please review again.

**Review by EricJoy2048 — COMMENTED** [review]
Row delimiter is need.

File: seatunnel-connectors-v2/connector-file/connector-file-base-hadoop/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/hdfs/source/BaseHdfsFileSource.java [hunk 2]
```diff
@@ -49,12 +51,26 @@ public void prepare(Config pluginConfig) throws PrepareFailException {
...
+                case JSON:
+                    Config schemaConfig = pluginConfig.getConfig(SeaTunnelSchema.SCHEMA);
+                    rowType = SeaTunnelSchema
+                            .buildWithConfig(schemaConfig)
+                            .getSeaTunnelRowType();
+                    readStrategy.setSeaTunnelRowTypeInfo(rowType);
+                    break;
+                case ORC:
+                case PARQUET:
+                    throw new UnsupportedOperationException("SeaTunnel does not support user-defined schema for [parquet, orc] files");
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-file/connector-file-base-hadoop/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/hdfs/source/BaseHdfsFileSource.java:69]: For Hive Source Reader, The hive columns num may greater than orc/parquet file columns num. So I think we need make orc/parquet file source support user-defined schema too.

**Review by EricJoy2048 — COMMENTED** [review]
Can we support read the sub path as partition fields ?

**Review by EricJoy2048 — COMMENTED** [review]
Can you add UT/E2E for this pr?

**Comment by TyrantLucifer** [comment]
> Can we support read the sub path as partition fields ?

Yes, but support this feature not in this pr, it will in #2985

File: seatunnel-connectors-v2/connector-file/connector-file-base-hadoop/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/hdfs/source/BaseHdfsFileSource.java (hunk 2 above)
Comment (TyrantLucifer) [reply seatunnel-connectors-v2/connector-file/connector-file-base-hadoop/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/hdfs/source/BaseHdfsFileSource.java:69]: When the hive connector upgrade is completed, this piece will be changed, what do you think?

**Comment by ashulin** [comment]
I rebase this PR onto dev.
@TyrantLucifer @EricJoy2048  CC

**Comment by TyrantLucifer** [comment]
> I rebase this PR onto dev. @TyrantLucifer @EricJoy2048 CC

Good job.

**Comment by CalvinKirs** [comment]
Overall good, can you add the change log to the corresponding document

**Comment by TyrantLucifer** [comment]
> Overall good, can you add the change log to the corresponding document

This job I will do in next pr, in next pr I will add all the change log of the connectors that developed by me.

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/TextReadStrategy.java [hunk 3]
```diff
@@ -35,22 +41,65 @@
...
     }
 
     @Override
     public SeaTunnelRowType getSeaTunnelRowTypeInfo(HadoopConf hadoopConf, String path) {
-        return new SeaTunnelRowType(new String[]{TEXT_FIELD_NAME},
-                new SeaTunnelDataType<?>[]{BasicType.STRING_TYPE});
+        SeaTunnelRowType simpleSeaTunnelType = SeaTunnelSchema.buildSimpleTextSchema();
+        deserializationSchema = TextDeserializationSchema.builder()
+                .seaTunnelRowType(simpleSeaTunnelType)
+                .delimiter(String.valueOf('\002'))
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/TextReadStrategy.java:72]: Default delimiter should be `\001`?

Comment (TyrantLucifer) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/TextReadStrategy.java:72]: If user do not assign schema information in config file, connector will not split for every line in target file. For example, if file connector read text file that generated by hive and setting delimiter to `\001` if will split the whole line. So in order to do not split the line content I chosed to set delimter `\002`. `\002` is not a common delimiter for text file.

**Review by Hisoka-X — APPROVED** [review]
LGTM

**Review by ashulin — APPROVED** [review]

**Pull request merged**